src/app/visual/readonly-grid/readonly-grid.component.html

```diff
@@ -1,17 +1,32 @@
-<div
-  class="readonly-grid"
+<sky-docs-demo-control-panel-theme
+  (themeSettingsChange)="themeSettingsChange($event)"
 >
-  <sky-ag-grid-wrapper>
-    <ag-grid-angular
-      [gridOptions]="gridOptions"
-      [rowData]="gridData"
-      skyBackToTop
-    >
-    </ag-grid-angular>
-  </sky-ag-grid-wrapper>
-  <sky-infinite-scroll
-    [enabled]="hasMore"
-    (scrollEnd)="onScrollEnd()"
+</sky-docs-demo-control-panel-theme>
+
+<ng-container
+  *ngIf="
+    (themeSvc?.settingsChange | async)?.currentSettings?.theme?.name === 'modern';
+    then grid else grid
```

Comment:
This looks like a hack to get around the fact that the grid isn't updated when settings change as mentioned above. Re-evaluate it once you've addressed that issue.